pkg/roachprod/vm/aws/aws.go

```diff
@@ -672,6 +677,38 @@ func (p *Provider) Create(
 		return errors.Errorf("Please specify a valid region.")
 	}
 
+	// Auto-detect architecture from machine-type, check that it's compatible
+	// with the requested arch (if any), and stores the value in opts.Arch
+	machineTypeArch, err := p.getCPUArchFromMachineType(l, machineType, regions[0], NarrowCacheKey)
```

Comment:
We don't really need this validation step when `opts.Arch == ""`, right? Also, since it's only a validation step, it might be a good candidate for its own helper function.

Reply:
The value of `opts.Arch` is used in [runInstance()](https://github.com/cockroachdb/cockroach/blob/master/pkg/roachprod/vm/aws/aws.go#L1170) to figure out which AMI to use.

The previous logic in this function was to use a match on the machine-type, but by validating (when provided) and forcing (when left empty) this param, I can simply use `if opts.Arch == string(vm.ArchARM64)`.